### PR TITLE
Traction service unregister; consist updates.

### PR DESCRIPTION
**Pull request by balazsracz**
Adds the ability to unregister a train node from the train service.
(Does not do this automatically at this point due to bugs.)
Makes the consisting operations virtual so train node implementations
can override the default behavior.

### CODE REVIEW

File: src/openlcb/TractionTrain.cxx [hunk 1]
```diff
@@ -62,6 +62,12 @@ TrainNodeForProxy::TrainNodeForProxy(TrainService *service, TrainImpl *train)
     service_->register_train(this);
 }
 
+TrainNodeForProxy::~TrainNodeForProxy()
+{
+    // breaks unit tests due to bugs
```
Comment (bakerstu) [inline src/openlcb/TractionTrain.cxx:67]: Perhaps an @todo comment to come back and fix this?

File: src/openlcb/TractionTrain.cxx [hunk 2]
```diff
@@ -70,6 +76,12 @@ TrainNodeWithId::TrainNodeWithId(
     service_->register_train(this);
 }
 
+TrainNodeWithId::~TrainNodeWithId()
+{
+    // breaks unit tests due to bugs
```
Comment (bakerstu) [inline src/openlcb/TractionTrain.cxx:81]: Same.

File: src/openlcb/TractionTrain.hxx [hunk 3]
```diff
@@ -212,6 +212,8 @@ class TrainNodeForProxy : public TrainNode {
 public:
     TrainNodeForProxy(TrainService *service, TrainImpl *train);
 
+    ~TrainNodeForProxy();
```
Comment (bakerstu) [inline src/openlcb/TractionTrain.hxx:215]: Doxygen header.

File: src/openlcb/TractionTrain.hxx [hunk 4]
```diff
@@ -221,6 +223,8 @@ class TrainNodeWithId : public TrainNode {
 public:
     TrainNodeWithId(TrainService *service, TrainImpl *train, NodeID node_id);
 
+    ~TrainNodeWithId();
```
Comment (bakerstu) [inline src/openlcb/TractionTrain.hxx:226]: Doxygen header.

File: src/openlcb/TractionTrain.cxx (hunk 1 above)
Comment (balazsracz) [reply src/openlcb/TractionTrain.cxx]: done

File: src/openlcb/TractionTrain.cxx (hunk 2 above)
Comment (balazsracz) [reply src/openlcb/TractionTrain.cxx]: done

File: src/openlcb/TractionTrain.hxx (hunk 3 above)
Comment (balazsracz) [reply src/openlcb/TractionTrain.hxx:220]: done. Also backfilled the others.

File: src/openlcb/TractionTrain.hxx (hunk 4 above)
Comment (balazsracz) [reply src/openlcb/TractionTrain.hxx:239]: done

**Pull request merged**